code/obj/machinery/launcherloader.dm

```diff
@@ -329,6 +329,15 @@
 
 	var/list/destinations = list("Airbridge", "Cafeteria", "EVA", "Engine", "Disposals", "QM", "Catering", "MedSci", "Security") //These have to match the ones on the cargo routers for the routers to work.
 
+	connection_scan()
+		..()
+		if (ismap("COGMAP2"))
+			destinations = list("Arrivals","Catering","Disposals","Engine","Escape","Export","MedSci","Security","Trader","QM")
+		if (ismap("KONDARU"))
+			destinations = list("Catering","Disposal","Engineering","Export","Medbay","Mining","Research","Pod Bay","Security","QM")
```

Comment:
Do you mean the map dmm? This is here because it initializes its destinations every time it is built. If someone repairs the terminal or builds a new one on the map, this is required to set the correct destinations.

Reply:
i mean map settings.
the `destinations =` bit - it should do `destinations = global.map_settings.idk_maybe_barcode_destinations`

Reply:
see:
https://github.com/goonstation/goonstation/blob/14673c20a9b6b1ecd9ec8c326a60616e6c934be6/code/map.dm#L81